thirdeye/thirdeye-hadoop/src/main/java/com/linkedin/thirdeye/hadoop/aggregation/AggregationPhaseMapOutputKey.java

```diff
@@ -56,22 +64,45 @@ public long getTime() {
     // dimensions size
     dos.writeInt(dimensions.size());
     // dimension values
-    for (String dimension : dimensions) {
-      bytes = dimension.getBytes();
-      dos.writeInt(bytes.length);
-      dos.write(bytes);
+    for (int i = 0; i < dimensions.size(); i++) {
+      Object dimension = dimensions.get(i);
+      DimensionType dimensionType = dimensionTypes.get(i);
+
+      switch (dimensionType) {
+      case DOUBLE:
+        dos.writeDouble((double) dimension) ;
```

Comment:
We should be able to support conversion here right. There must be some method in Pinot to convert something from one datatype to another. We have two cases its either a String or Number. If its Number then use the right method to convert the data to appropriate type https://docs.oracle.com/javase/7/docs/api/java/lang/Number.html. If it's String always use the Integer.parseInt or Long.parseLong based on the data type. This will handle cases where each avro file might have a different data type for the same input column